app/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -4,239 +4,325 @@
-                (irq_callbacks_##n[i].work = &data->work; irq_callbacks_##n[i].dev = dev;          \
-                 irq_callbacks_##n[i].debounce_period = cfg->debounce_period;                      \
-                 gpio_init_callback(&irq_callbacks_##n[i].callback,                                \
-                                    kscan_gpio_irq_callback_handler, BIT(in_cfg->pin));            \
-                 err = gpio_add_callback(input_devices[i], &irq_callbacks_##n[i].callback);        \
-                 if (err) {                                                                        \
-                     LOG_ERR("Error adding the callback to the column device");                    \
-                     return err;                                                                   \
-                 }),                                                                               \
-                ())                                                                                \
-        }                                                                                          \
-        data->dev = dev;                                                                           \
-        COND_CODE_1(IS_ENABLED(CONFIG_ZMK_KSCAN_DIRECT_POLLING),                                   \
-                    (k_timer_init(&data->poll_timer, kscan_gpio_timer_handler, NULL);), ())        \
-        if (cfg->debounce_period > 0) {                                                            \
-            k_work_init_delayable(&data->work.delayed, kscan_gpio_work_handler);                   \
-        } else {                                                                                   \
-            k_work_init(&data->work.direct, kscan_gpio_work_handler);                              \
-        }                                                                                          \
-        return 0;                                                                                  \
-    }                                                                                              \
-    static const struct kscan_gpio_config kscan_gpio_config_##n = {                                \
-        .inputs = {UTIL_LISTIFY(INST_INPUT_LEN(n), KSCAN_DIRECT_INPUT_ITEM, n)},                   \
-        .num_of_inputs = INST_INPUT_LEN(n),                                                        \
-        .debounce_period = DT_INST_PROP(n, debounce_period)};                                      \
-    DEVICE_DT_INST_DEFINE(n, kscan_gpio_init_##n, NULL, &kscan_gpio_data_##n,                      \
-                          &kscan_gpio_config_##n, POST_KERNEL, CONFIG_ZMK_KSCAN_INIT_PRIORITY,     \
-                          &gpio_driver_api);
-
-DT_INST_FOREACH_STATUS_OKAY(GPIO_INST_INIT)
+#define KSCAN_DIRECT_INIT(index)                                                                   \
```

Comment:
The usual convention for driver is either `n` or `inst` here. I'd say we should stick w/ that convention unless we have a real reason to differ.

Reply:
Done. Also updated the matrix driver to match.